src/content/guides/creating-your-own-admission-controller/index.md

```diff
@@ -0,0 +1,219 @@
++++
+title = "Creating your own admission controller"
+description = "Tutorial on how to deploy Istio on a Giant Swarm Kubernetes cluster."
+date = "2018-10-26"
+type = "page"
+weight = 50
+tags = ["tutorial"]
++++
+
+# Creating your own admission controller
+
+The Kubernetes API is an amazing territory. Thanks to being built around the REST model, it gives us the possibility to manage all our workloads using HTTP requests. Tools like `kubectl` or `Kubernetes dashboard` take advantage of this, helping to manage the different resources. But Kubernetes API is far more. Let's take a deeper look at how it is composed:
+
+![](./src/static/img/api_components.png)
+
+The picture highlights the different components living inside the API component. The request starts the API journey interfacing with the authentication controller. Once the request is authenticated, the authorization module dictates if the request issuer may perform the operation. After the request is properly authorized, the admission magic comes into play. 
+
+There are two types of admission controllers in Kubernetes. They work slightly differently. The first one, it is the **validating admission controller**, which proxies the requests to the subscribed webhooks. The Kubernetes API registers the webhooks based on the resource type and the request method. Every webhook runs some logic to validate the incoming resource and it replies to the API with a verdict. In case the validation webhook rejects the request, Kubernetes API returns a failed HTTP response to the user. Otherwise, it continues to the next admission.
+
+In the second, there is a **mutation admission controller** which modifies the resource submitted by the user, so you can make some defaults or force some schema. The cluster admins can attach mutation webhooks to the API to be run in the same way as validation. Indeed mutation logic runs one step before the validation.
+
+__Note:__ I would like to mention that Kubernetes API allows you to register custom resource objects too, called `Custom Resource Definition`. In addition, you can create an API extension server which listens to a new REST path. But in this guide, we focus exclusively on the admission functionality.
+
+## Goal
+
+Our goal here is to create a simple validation controller which will empower us to influence the pod creation. Although there are many more possibilities and the logic can be as complex as you need. The goal is to create a basic version which makes a simple validation. You can find more real examples in the links at the bottom.
+
+Our example controller will be called `grumpy` and will reject all new pods with a name different than `smooth-app`. I recognize it may be tempting to deploy this controller in a real cluster ;).
```

Comment:
we refrain from using first person in docs, as there's no author information here

Reply:
```suggestion
Our example controller will be called `grumpy` and will reject all new pods with a name different than `smooth-app`. We recognize it may be tempting to deploy this controller in a real cluster ;).
```